Cython/Utility/CppConvert.pyx

```diff
@@ -39,11 +39,16 @@ cdef inline object {{cname.replace("PyObject", py_type, 1)}}(const string& s):
 
 cdef extern from *:
     cdef cppclass vector "std::vector" [T]:
-        void push_back(T&)
+        ctypedef size_t size_type
+
+        void reserve(size_type) except +
```

Comment:
Why the typedef?

Reply:
To match the definition in the vector.pxd file. It could be `        void reserve(size_t) except +` instead if that is preferred

Reply:
Is there a reason, why I can not directly import the pxd files for these containers here, or did they simply not exist back then and should be used in there as well?

Reply:
These declarations are small and self-contained, and they avoid a bit of pxd lookup overhead when processing the snippets. Given that users can modify the import path, it seems safer to keep them right in here, in order to avoid potential failures that would be rather difficult to understand for users.

Reply:
makes sense